### PR TITLE
Added Hyperlink Button to URL preview

### PR DESCRIPTION
**Pull request by anoverga17**
## Motivation and Context

This pull request adds an important add on to the url uploader, that is the ability to see links uploaded from a markusurl file and be able to visit the link in a new tab. To do this, minor changes were also made to the layout of the `url_viewer`

## Your Changes

**Description**:
- Added a bar that shows a clickable text of the entire URL. Clicking the link opens it in a new page.   
- Reworked layout of `url_viewer` to be adjustable according to the length of the url.
- Fixed minor bug that did not reset url modal content after submitting
- Added error messages for unsupported and/or broken URLs

**Type of change** (select all that apply):

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (internal change to codebase, without changing functionality)

## Testing

Manual testing of the feature was done to ensure no additional features were broken by the `url_viewer` rework. Several types of links were uploaded and viewed to ensure that the the proper messages were being displayed and the appropriate link was shown. 

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have verified that the pre-commit.ci checks have passed. 
- [x] I have verified that the CI tests have passed. 
- [x] I have reviewed the test coverage changes reported on Coveralls.

### CODE REVIEW

File: app/assets/javascripts/Components/Modals/submission_url_submit_modal.jsx [hunk 1]
```diff
@@ -16,6 +16,10 @@ class SubmitUrlUploadModal extends React.Component {
 
   onSubmit = event => {
     event.preventDefault();
+    this.setState({
```
Comment (mishaschwartz) [inline app/assets/javascripts/Components/Modals/submission_url_submit_modal.jsx:19]: `setState` is asynchronous meaning it _might_ update the state after the `this.props.onSubmit` is triggered but then again it might update it before, and then you'll end up passing empty strings as arguments to that function.

File: app/assets/javascripts/Components/Result/url_viewer.jsx [hunk 2]
```diff
@@ -5,15 +5,16 @@ export class URLViewer extends React.Component {
...
   }
 
   componentDidMount() {
     this.configDisplay();
   }
 
-  componentDidUpdate(prevProps) {
-    if (prevProps.externalUrl !== this.props.externalUrl) {
+  componentDidUpdate(prevProps, prevState) {
+    if (prevProps !== this.props) {
```
Comment (mishaschwartz) [inline app/assets/javascripts/Components/Result/url_viewer.jsx:17]: Javascript objects are not equal as long as they don't refer to the same object in memory. This means that this if statement may be true even if the contents of `prevProps` and `this.props` are identical. If you're doing this comparison, it is safer to compare the content of the props objects, not the objects themselves.

File: app/assets/javascripts/Components/Result/url_viewer.jsx [hunk 3]
```diff
@@ -5,15 +5,16 @@ export class URLViewer extends React.Component {
...
     };
   }
 
   componentDidMount() {
     this.configDisplay();
   }
 
-  componentDidUpdate(prevProps) {
-    if (prevProps.externalUrl !== this.props.externalUrl) {
+  componentDidUpdate(prevProps, prevState) {
```
Comment (mishaschwartz) [inline app/assets/javascripts/Components/Result/url_viewer.jsx:16]: don't include prevState as an argument if the function doesn't use it.

File: app/assets/javascripts/Components/Result/url_viewer.jsx [hunk 4]
```diff
@@ -5,15 +5,16 @@ export class URLViewer extends React.Component {
     super(props);
     this.state = {
       url: "",
```
Comment (mishaschwartz) [inline app/assets/javascripts/Components/Result/url_viewer.jsx:7]: Do we need to store the url in state anymore? Can't we just use the `this.props.externalUrl` value instead?

File: config/locales/views/results/en.yml [hunk 5]
```diff
@@ -5,7 +5,7 @@ en:
       across_all_submission_files: Click on the filename to jump to the annotation.
       common: Common Annotations
       include_in_download: Include Annotations
-      include_in_download_warning: 'Warning: Annotations on PDFs are not currently downloaded.'
+      include_in_download_warning: 'Warning: Annotations on PDFs and URLs are not currently downloaded.'
```
Comment (mishaschwartz) [inline config/locales/views/results/en.yml:8]: This isn't necessary here. We haven't enabled any sort of annotations on URLs yet so this warning is misleading. Please revert this change.

**Review by mishaschwartz — CHANGES_REQUESTED** [review]

File: app/assets/javascripts/Components/Result/url_viewer.jsx (hunk 4 above)
Comment (anoverga17) [reply app/assets/javascripts/Components/Result/url_viewer.jsx:7]: We can. This was a design decision though since `this.props.externalUrl` can be a string that says the file cannot be loaded. I thought therefore it was cleaner to store urls specifically in the state since other ways I thought of of flagging an unreadable URL wasn't as readable.

**Review by mishaschwartz — APPROVED** [review]
